contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -638,4 +644,52 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     uint32 configCount = s_dons[donId].configCount;
     return s_dons[donId].config[configCount].capabilityConfigs[capabilityId];
   }
+
+  /// @notice Sets the configuration for a DON
+  /// @param donId The ID of the DON to set the configuration for
+  /// @param configCount The number of times the DON has been configured
+  /// @param nodes The nodes making up the DON
+  /// @param capabilityConfigurations The list of configurations for the
+  /// capabilities supported by the DON
+  /// @param isPublic True if the DON is can accept external workflows
+  function _setDONConfig(
+    uint32 donId,
+    uint32 configCount,
+    bytes32[] calldata nodes,
+    CapabilityConfiguration[] calldata capabilityConfigurations,
+    bool isPublic
+  ) internal {
+    s_dons[donId].isPublic = isPublic;
+
+    DONCapabilityConfig storage donCapabilityConfig = s_dons[donId].config[configCount];
+    for (uint256 i; i < nodes.length; ++i) {
```

Comment:
Thinking a bit on how to optimize this.  Currently the code will update the DON's nodes and supported capability configs even if the function parameters are the same as what's already in storage.  One idea I had was to store the hash of `nodes` and `capabilityConfigs` and compare them against the hash of the params passed in.

Reply:
Create a ticket to test this out during the gas optimization stage?